src/components/Nav/BackgroundDropdownItem.tsx

```diff
@@ -19,9 +22,39 @@ export const BackgroundDropdownItem = ({
         "cursor-pointer bg-gray-300 py-1 hover:bg-gray-100 dark:bg-gray-800 dark:hover:bg-gray-900",
         isPicked && "bg-gray-100 dark:bg-gray-900"
       )}
-      onClick={() => setIsBackground(background)}
+      onClick={() => setBackgroundId(background)}
     >
-      <div className="block px-4 py-2 text-sm text-gray-700 dark:text-gray-200">{title}</div>
+      {title === "Custom Color" ? (
+        <CustomColorPicker />
+      ) : (
+        <div className="block px-4 py-2 text-sm text-gray-700 dark:text-gray-200">{title}</div>
+      )}
+    </div>
+  );
+};
+
+const colors = ["#000000", "#383838", "#654724", "#312465", "#1F6353", "#652424"];
+const CustomColorPicker = () => {
+  const { backgroundId, backgroundColor, setBackgroundColor } = useSetBackground();
+  const isUsingCustomBackground = backgroundId === Background.CUSTOM_COLOR;
+
+  return (
+    <div>
+      <div className="block py-2 pl-4 text-sm text-gray-700 dark:text-gray-200">Color</div>
+      <div className="ml-4 flex pb-2">
+        {colors.map(col => (
```

Comment:
You are missing a unique key prop here.
LMK if you want to huddle on this so i can explain the concept :)
More than happy